templates/haproxy.cfg

```diff
@@ -52,9 +52,19 @@ frontend fe_http
 
 frontend fe_https
     bind *:443 ssl crt {{ LOAD_BALANCER_CERTS_DIR }}
+    acl fail_mode status 500
+    acl fail_mode status 502
+    acl fail_mode status 504
+    errorloc 500 /maintenance
```

Comment:
I find it confusing that we're using the term "maintenance" for 500 errors, and that we're even putting it in the path. I think it would be much easier for us and our users to understand if we renamed this backend to `server-error` and/or changed the new URL path to `/error`.

I also want to update `index.html` to use different wording, but that wouldn't be in this PR.

Reply:
I think the alternative to putting it in the path is to add the maintenance/error backend as a backup backend and have the health check fail in case of a 50x error. I think the maintenance name has stuck around because this ticket has oscillated between being about specifically considering maintenance pages for one instance, and between creating a generic solution that works for everyone.

Reply:
Also, about updating ``index.html`` to use different wording, there is a ticket for that OC-3404. I am also working on that one as well.

Reply:
@xitij2000 I don't think the fallback server idea works well – see my reply on the ticket.  Using `/error` instead of (or in addition to `/maintenance` would work of course.

Reply:
@smarnach Sure. Will rename it to ``/error`` then. I don't think ``/error*`` is used by Open edX itself so it should be OK.